src/lib/bmp/bmp.c

```diff
@@ -0,0 +1,226 @@
+/**
+ * @file
+ * @brief lib for read and draw BMP
+ *
+ * @date Apr 18, 2019
+ * @author Filipp Chubukov
+ */
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/ioctl.h>
+
+#include <linux/fb.h>
+
+/* now it works with non pallete bmp formats - 24 and 32 bits, tested on 32bpp fb and not tested on 16bpp fb */
+typedef struct {
+	unsigned short bfType;	      /* == "BM" */
```

Comment:
`unsigned short`, `unsigned int` and even `char` are platform-dependent (i.e. they are not guaranteed to be 16 and 32 bit wide on any platform), so `uint16_t` and `uint32_t` are preferable to them.

Reply:
bmp lib was fully reworked